### PR TITLE
Adds the carets to the preview and form

### PR DESCRIPTION
**Pull request by jcomack**
Also attempted to add tests, but this does not seem to be working properly.

### CODE REVIEW

File: composites/SearchResultEditor/SearchResultEditor.js [hunk 1]
```diff
@@ -167,7 +226,51 @@ class SearchResultEditor extends React.Component {
...
+					focusedFormField: formField,
+				} );
+				break;
+			case "mouseenter":
+				this.setState( {
+					hoveredPreviewField: previewField,
+					hoveredFormField: formField,
+				} );
+				break;
+
```
Comment (terw-dan) [inline composites/SearchResultEditor/SearchResultEditor.js]: Unnecessary white line.

File: composites/SearchResultPreview/SearchResultPreview.js [hunk 2]
```diff
@@ -89,23 +158,33 @@ class SearchResultPreview extends React.Component {
...
+                    <span ref="previewTitle" className={this.classNames.title}>{translations.title}</span>
 				</div>
 
-				<div className="yoast-search-result-preview__field">
-					<span className="screen-reader-text">{translations.urlLabel}</span>
-					<span ref="previewUrl" className="yoast-search-result-preview__url">{this.props.url}</span>
+				<div ref="previewUrlContainer"
+	                 className={this.setClassNameForField( "previewUrlContainer" )}
+	                 {...this.mouseEventHandlers( "previewUrlContainer", "formSlug" )}>
+					<span className={this.classNames.screenReaderText}>{translations.urlLabel}</span>
```
Comment (terw-dan) [inline composites/SearchResultPreview/SearchResultPreview.js]: strange alignment

File: composites/SearchResultForm/SearchResultForm.js [hunk 3]
```diff
@@ -19,22 +19,31 @@ class SearchResultForm extends React.Component {
...
+			button: "yoast-button yoast-search-result-form__close-button",
+		};
 	}
 
 	/**
 	 * Retrieves translatable strings to be used throughout the component.
-	 * 
-	 * @returns {{formTitle: string, formDescription: string, titleLabel: string, title: string, urlLabel: string, descriptionLabel: string, description: string}}
+	 *
+	 * @returns {object} An object containing all the possible translations.
```
Comment (atimmer) [inline composites/SearchResultForm/SearchResultForm.js]: `object` should have a capital.

**Comment by terw-dan** [comment]
CR: can you check the whitespaces again, they aren't consistent through out all files.

**Comment by atimmer** [comment]
We can iterate on this further on the `develop` branch.

**Pull request merged**